### PR TITLE
ente-auth: init at 3.1.1

### PR DESCRIPTION
**Pull request by schnow265**
## Description of changes

An end-to-end encrypted, cross platform and free app for storing your 2FA codes with cloud backups. Works offline. You can even use it without signing up for an account if you don't want the cloud backups or multi-device sync.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Supernovatux** [comment]
Why is this package not built from source?

**Comment by niklaskorz** [comment]
This is missing a .desktop file and launching from the command line fails (using proprietary nvidia drivers on Wayland):

```
(ente_auth:126207): Gdk-CRITICAL **: 12:13:56.123: gdk_gl_context_make_current: assertion 'GDK_IS_GL_CONTEXT (context)' failed

** (ente_auth:126207): WARNING **: 12:13:56.123: Failed to initialize GLArea: No GL implementation is available

** (ente_auth:126207): WARNING **: 12:13:56.124: gtk.dart: failed to call method: No engine to send to

** (ente_auth:126207): WARNING **: 12:13:56.127: gtk.dart: failed to call method: No engine to send to
```

File: pkgs/by-name/en/ente-auth/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+in
+
+appimageTools.wrapType2 {
+  inherit pname version src;
+  extraPkgs = pkgs: [
+    pkgs.libepoxy
+    pkgs.brotli
+  ];
+
+  meta = {
```
Comment (niklaskorz) [inline pkgs/by-name/en/ente-auth/package.nix:23]: Missing `platforms = [ "x86_64-linux" ];` (as the provided appimage only supports x86_64-linux)

File: pkgs/by-name/en/ente-auth/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  };
+in
+
+appimageTools.wrapType2 {
+  inherit pname version src;
+  extraPkgs = pkgs: [
+    pkgs.libepoxy
+    pkgs.brotli
+  ];
+
```
Comment (niklaskorz) [inline pkgs/by-name/en/ente-auth/package.nix:22]: To install the desktop file:

```nix
let
  # ...
  appimageContents = appimageTools.extractType1 { inherit pname version src; };
in
appimageTools.wrapType2 {
  # ...

  extraInstallCommands = ''
    install -m 444 -D ${appimageContents}/ente_auth.desktop -t $out/share/applications
    substituteInPlace $out/share/applications/ente_auth.desktop \
      --replace-fail 'Exec=LD_LIBRARY_PATH=usr/lib ente_auth' 'Exec=ente-auth'
    cp -r ${appimageContents}/usr/share/icons $out/share
  '';
  
  # ...
}
```

File: pkgs/by-name/en/ente-auth/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+  appimageTools,
+  fetchurl,
+  lib,
+}:
+let
+  pname = "ente-auth";
+  version = "3.0.18-beta.2";
```
Comment (niklaskorz) [inline pkgs/by-name/en/ente-auth/package.nix:8]: Any reason this is packaging a beta instead of the latest release `3.1.1`?

Comment (schnow265) [reply pkgs/by-name/en/ente-auth/package.nix]: Yes, as when I made this PR ``3.0.18-beta.2`` was the latest available version.

File: pkgs/by-name/en/ente-auth/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      --replace-fail 'Exec=LD_LIBRARY_PATH=usr/lib ente_auth' 'Exec=ente-auth'
+    cp -r ${appimageContents}/usr/share/icons $out/share
+  '';
+
+  extraPkgs = pkgs: [
+    pkgs.libepoxy
+    pkgs.brotli
+  ];
+
+  platforms = [ "x86_64-linux" ];
```
Comment (niklaskorz) [inline pkgs/by-name/en/ente-auth/package.nix:33]: This belongs inside the meta attribute set

File: pkgs/by-name/en/ente-auth/package.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  fetchurl,
+  lib,
+}:
+let
+  pname = "ente-auth";
+  version = "3.1.1";
+
+  src = fetchurl {
+    url = "https://github.com/ente-io/ente/releases/download/auth-v${version}/ente-auth-v${version}-x86_64.AppImage";
+    hash = "sha256-WtFoqW++U+BoDUW1lc6xMJDCC1r2+O/fci492P2iVDg=";
```
Comment (niklaskorz) [inline pkgs/by-name/en/ente-auth/package.nix:12]: error: hash mismatch in fixed-output derivation '/nix/store/xvkjfr0bw2322p2bznw772n9y0c3wn8d-ente-auth-v3.1.1-x86_64.AppImage.drv':
         specified: sha256-WtFoqW++U+BoDUW1lc6xMJDCC1r2+O/fci492P2iVDg=
            got:    sha256-RcH+1YM6gPneVbv158LejClkrCNnAwr6K5hKbcyvTMI=

**Review by niklaskorz — APPROVED** [review]

**Comment by schnow265** [comment]
Closing in favor of #339059.

**Pull request closed without merging**